src/components/avatar.rs

```diff
@@ -112,49 +114,68 @@ impl Avatar {
         glib::Object::new()
     }
 
-    fn setup_expressions(&self) {
-        let item_expression = Self::this_expression("item");
+    fn create_signal_groups(&self) {
```

Comment:
There is way too much `if else` branching in the rest of the file hurting readability. I'd like to simplify that by giving it a clearer structure.
I'm going to make some suggestion.